### ISSUE TITLE
Oracle and Wyvern ulti interaction

### ISSUE THREAD

**Issue by eldarium** [jeffhill/Dota2#597]
If you ulti someone as Oracle who's under Winter's Curse, the target will get denied if there was enough damage. Seen in match 6602731842 at 19:58

**Comment by Dopwop** [comment]
Weird interaction, but it's consistent with both abilities so it's probably the intended outcome.
Winter Wyvern gets the kill if the target dies under Winter's Curse, Winter's Curse has ended so Wyvern doesn't get the kill.
False Promise still has damage to deal out in order, the source of damage is a teammate, so Sven denies Windrunner.

![FalsePromise+WintersCurse-1](https://user-images.githubusercontent.com/106779797/172051370-b9ffa95c-bbc5-4ea1-b38a-0a0adbdda638.png)

**Comment by eldarium** [comment]
I agree the behaviour might be technically correct, but at the other hand - rules for denying, especially heroes, are strictly defined. I don't even think you could deny your ally with any other spell, even under WV ulti (like e.g. Veno's passive). It seems there should be a damage source check if it's a code oversight, or probably it's on the realm of ice frog (i.e. Balancing)

**Comment by shaileshgarg** [comment]
False promise can lead to a deny, during false promise all heal is summed and doubled then damage is reduced from it, whichever heros damage reduces it below zero, gets the kill. If this happens due to ally damage,  then the hero is denied.

This is intended interaction. Can be closed.
